packages/core-flows/src/definition/cart/workflows/create-payment-collection-for-cart.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  CartDTO,
+  CreatePaymentCollectionForCartWorkflowInputDTO,
+} from "@medusajs/types"
+import {
+  WorkflowData,
+  createWorkflow,
+  transform,
+} from "@medusajs/workflows-sdk"
+import { retrieveCartStep } from "../steps"
+import { createPaymentCollectionsStep } from "../steps/create-payment-collection"
+import { linkCartAndPaymentCollectionsStep } from "../steps/link-cart-payment-collection"
+
+export const createPaymentCollectionForCartWorkflowId =
+  "create-payment-collection-for-cart"
+export const createPaymentCollectionForCartWorkflow = createWorkflow(
+  createPaymentCollectionForCartWorkflowId,
+  (
+    input: WorkflowData<CreatePaymentCollectionForCartWorkflowInputDTO>
+  ): WorkflowData<CartDTO> => {
+    const created = createPaymentCollectionsStep([input])
+
+    const link = transform({ cartId: input.cart_id, created }, (data) => ({
+      links: [
+        {
+          cart_id: data.cartId,
+          payment_collection_id: data.created[0].id,
+        },
+      ],
+    }))
+
+    linkCartAndPaymentCollectionsStep(link)
```

Comment:
```suggestion
    const linksInput = {
      links: [
        {
          cart_id: input.cart_id,
          payment_collection_id: created[0].id,
        },
      ],
    }

    linkCartAndPaymentCollectionsStep(links)
```

Reply:
Am I wrong in thinking this is not possible? The `created[0].id` is not resolved yet